### PR TITLE
Scenario Service: Use an unique lf version for all the loaed modules.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/scenario-service/protos/scenario_service.proto [hunk 1]
```diff
@@ -104,6 +104,11 @@ message UpdateContextRequest {
   UpdateModules update_modules = 2;
   UpdatePackages update_packages = 3;
   bool noValidation = 4; // if true, does not run the package validations
+  // The lf v1 minor of the modules in `load_packages`
+  // The empty sting means no changed.
```
Comment (remyhaemmerle-da) [inline compiler/scenario-service/protos/scenario_service.proto:108]:
```suggestion
  // The empty sting means no change of version.
```

File: compiler/scenario-service/protos/scenario_service.proto [hunk 2]
```diff
@@ -104,6 +104,11 @@ message UpdateContextRequest {
   UpdateModules update_modules = 2;
   UpdatePackages update_packages = 3;
   bool noValidation = 4; // if true, does not run the package validations
+  // The lf v1 minor of the modules in `load_packages`
+  // The empty sting means no change of version.
+  // Can be changed from one call to the other only if all the previously
```
Comment (cocreature) [inline compiler/scenario-service/protos/scenario_service.proto:109]: Being able to change this seems like a fair amount of complexity for no gain afaict. Why not specify the LF version as a CLI parameter to the scenario service? Afaict, we can fix that statically in the compiler and we definitely don’t change it during runtime.

Comment (remyhaemmerle-da) [reply compiler/scenario-service/protos/scenario_service.proto]: redesigned.

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Context.scala [hunk 3]
```diff
@@ -86,16 +87,13 @@ class Context(val contextId: Context.ContextId) {
...
-      major: LanguageVersion.Major,
-      minor: String,
-      bytes: ByteString,
-  ): Ast.Module = {
-    val lfVer = LanguageVersion(major, LanguageVersion.Minor fromProtoIdentifier minor)
-    val dop: Decode.OfPackage[_] = Decode.decoders
-      .lift(lfVer)
-      .getOrElse(throw Context.ContextException(s"No decode support for LF ${lfVer.pretty}"))
-      .decoder
+  private[this] val dop: Decode.OfPackage[_] = Decode.decoders
```
Comment (cocreature) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Context.scala:90]: döp döp döp https://www.youtube.com/watch?v=0aV_vHcunSQ

**Pull request merged**